server/auth/auth.py

```diff
@@ -21,6 +16,11 @@ def requires_client_login(self):
         """Return True if the user needs to login from the client (e.g. Login button is shown)"""
         return False
 
+    @abstractmethod
+    def complete_setup(self, app):
```

Comment:
Could there be a better name for this function or at least a better docstring? I have very little understanding of what is supposed to happen in this function if I were to use this interface. What are the things that are necessary to "complete setup"?

Reply:
I added to the docstring.  This is just a hook that gets called before the server starts running.   It's entirely up to the authentication type to do something that makes sense here.   The oauth creates a client to the authentication service here, and it's a no-op for the rest of the classes.